lib/spack/spack/mirror.py

```diff
@@ -154,8 +154,68 @@ def push_url(self):
         """Get the valid, canonicalized fetch URL"""
         return self.get_url("push")
 
+    def validate(self, direction="push"):
+        access_pair = self._get_value("access_pair", direction)
+        access_token_variable = self._get_value("access_token_variable", direction)
+
+        errors = []
+
+        # Verify that the credentials that are variables expand
+        if access_pair and isinstance(access_pair, dict):
+            if "id_variable" in access_pair:
+                if access_pair["id_variable"] not in os.environ:
```

Comment:
have you heard of `and` ;)

Reply:
The original line was too long :p I can make it an `and` now